src/api/operations.ts

```diff
@@ -225,33 +225,85 @@ export const drawRectangle = (options: {
   graphicsState?: string | PDFName;
   matrix?: TransformationMatrix;
   clipSpaces?: Space[];
-}) => [
-    pushGraphicsState(),
-    options.graphicsState && setGraphicsState(options.graphicsState),
-    options.color && setFillingColor(options.color),
-    options.borderColor && setStrokingColor(options.borderColor),
-    setLineWidth(options.borderWidth),
-    options.borderLineCap && setLineCap(options.borderLineCap),
-    setDashPattern(options.borderDashArray ?? [], options.borderDashPhase ?? 0),
-    ...(options.clipSpaces ? clipSpaces(options.clipSpaces) : []),
-    options.matrix && concatTransformationMatrix(...options.matrix),
-    translate(options.x, options.y),
-    rotateRadians(toRadians(options.rotate)),
-    skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
-    moveTo(0, 0),
-    lineTo(0, options.height),
-    lineTo(options.width, options.height),
-    lineTo(options.width, 0),
-    closePath(),
-
-    // prettier-ignore
-    options.color && options.borderWidth ? fillAndStroke()
+  radius?: number | PDFNumber;
+}) => {
+  if (!options.radius || asNumber(options.radius) <= 0) {
+    return [
```

Comment:
There are many redundancies, so please write it more efficiently.

Reply:
The `drawRectangle` function is divided into two large blocks due to conditional branching, resulting in significant code duplication.  

You can extract the common parts to make the function more DRY.